stumpy/core.py

```diff
@@ -2492,7 +2492,7 @@ def _get_partial_mp_func(mp_func, client=None, device_id=None):
     elif device_id is not None:
         partial_mp_func = functools.partial(mp_func, device_id=device_id)
     else:
-        partial_mp_func = mp_func
+        partial_mp_func = functools.partial(mp_func)
```

Comment:
I made the change above so that the output of this functions always has type `functools.partial`. Otherwise, in function `core._compute_P_ABBA`, where we have:

```
# in core._compute_P_ABBA
...
partial_mp_func = _get_partial_mp_func(mp_func, client=client, device_id=device_id)
if inspect.signature(partial_mp_func).parameters.get("normalize") is not None:
        params = partial_mp_func.keywords
```

we first need to check if `partial_mp_func` is of type `function` or `functools.partial`, and if it is the former, we cannot use the attribute `.keywords `. 

What do you think @seanlaw ? If you are not okay with this, then we can just add a code (like this one: https://stackoverflow.com/questions/12627118/get-a-function-arguments-default-value) to get default value of argument "subseq_isconstant". Or, we can initialize them with `None`, and get their value only if `partial_mp_func` is of type `functools.partial`.

Reply:
@seanlaw 
We are calling `_get_partial_mp_func` in `stumpy.stimp`, `stumpy.ostinato`, and their non-normalize versions. 

https://github.com/TDAmeritrade/stumpy/blob/814977f110403bcfd1629bcf1ba4f90938394bfe/stumpy/stimp.py#L188-L191

we may just change `self._mp_func` to `self._partial_mp_func`

---

https://github.com/TDAmeritrade/stumpy/blob/814977f110403bcfd1629bcf1ba4f90938394bfe/stumpy/ostinato.py#L228-L239

Reply:
> we may just change self._mp_func to self._partial_mp_func

Yeah, I think that would be a good idea as it would be slightly more consistent